axelrod/result_set.py

```diff
@@ -633,3 +643,90 @@ def csv(self):
                     for rank in self.ranking]
             writer.writerow(list(map(str, data)))
         return csv_string.getvalue()
+
+
+class ResultSetFromFile(ResultSet):
+    """A class to hold the results of a tournament.
+
+    Initialised by a csv file of the format:
+
+
+    [p1index, p2index, p1name, p2name, p1rep1ac1p2rep1ac1p1rep1ac2p2rep1ac2,
+    ...]
+    [0, 1, Defector, Cooperator, DCDCDC, DCDCDC, DCDCDC,...]
+    [0, 2, Defector, Alternator, DCDDDC, DCDDDC, DCDDDC,...]
+    [1, 2, Cooperator, Alternator, CCCDCC, CCCDCC, CCCDCC,...]
+    """
+
+    def __init__(self, filename, with_morality=True):
+        """
+        Parameters
+        ----------
+            filename : string
+                name of a file of the correct file.
+            with_morality : bool
+                a flag to determine whether morality metrics should be
+                calculated.
+        """
+        self.players, self.interactions = self._read_csv(filename)
+        self.nplayers = len(self.players)
+        self.nrepetitions = len(self.interactions)
+
+        # Calculate all attributes:
+        self.build_all(with_morality)
+
+    def _read_csv(self, filename):
+        """
+        Returns
+        -------
+
+            A tuple:
+                - First element: list of player names
+                - Second element: interactions (list of dictionaries mapping
+                  index indices to interactions)
+        """
+        players_d = {}
+        interactions_d = {}
+        with open(filename, 'r') as f:
+            for row in csv.reader(f):
+                index_pair = (int(row[0]), int(row[1]))
```

Comment:
Can you include some comments about the structure of the csv flie -- i.e. what is in `row[0]`, `row[1]`, etc.